pkg/trie/triedb/cache_bench_test.go

```diff
@@ -98,13 +97,10 @@ func Benchmark_NodesCache(b *testing.B) {
 		}
 	})
 
-	// TODO: we still have some room to improve here, we are caching the raw
-	// node data and we need to decode it every time we access it. We could
-	// cache the decoded node instead and avoid decoding it every time.
+	// This is the same as iterate_all_entries_without_cache since the raw iterator calls TrieDB.getNodeOrLookup
 	b.Run("iterate_all_entries_with_cache", func(b *testing.B) {
```

Comment:
No the iterator isn't using the cache.  I've mirrored how they do it in substrate in this case.  I think it's because iterating to the next key use case isn't all that common, and it doesn't make sense to cache all those values for the regular read access pattern.  Given that the caches are implemented as LRUs, it will evict all cache entries if we supply the cache when iterating over all the keys.

Reply:
Got it!